### PR TITLE
bpo-31070: Fix a race condition in importlib _get_module_lock().

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue31070

### CODE REVIEW

File: Lib/importlib/_bootstrap.py [hunk 1]
```diff
@@ -172,8 +172,13 @@ def _get_module_lock(name):
                 lock = _DummyModuleLock(name)
             else:
                 lock = _ModuleLock(name)
-            def cb(_):
-                del _module_locks[name]
+            def cb(ref, name=name):
+                _imp.acquire_lock()
+                try:
+                    if _module_locks.get(name) is ref:
```
Comment (vstinner) [inline Lib/importlib/_bootstrap.py]: I suggest to add a comment on the test. Something like:

```
# bpo-31070: Check if another thread created a new lock
# after the previous lock was destroyed
# but before the weakref callback was called
```

**Review by vstinner — APPROVED** [review]
LGTM, I just suggest to add a comment.

File: Lib/importlib/_bootstrap.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/importlib/_bootstrap.py]: Done.

**Review by vstinner — APPROVED** [review]
LGTM, thanks for the comment.

**Review by ncoghlan — APPROVED** [review]
Thanks @serhiy-storchaka!

**Pull request merged**